src/zenml/orchestrators/step_launcher.py

```diff
@@ -173,64 +168,85 @@ def launch(self) -> None:
                 )
 
         try:
-            if run_was_created:
-                pipeline_run_metadata = self._stack.get_pipeline_run_metadata(
-                    run_id=pipeline_run.id
-                )
-                publish_utils.publish_pipeline_run_metadata(
+            with redirect_stdout(
+                step_logging.StepStdOut()
```

Comment:
Another silly question: if we can redirect stdout/stderr like this and syphon everything into the artifact store, why do we even need to manage everything as a Python logger ? After all, logs are also printed to stdout, right ?